tlsutil/config.go

```diff
@@ -857,10 +857,23 @@ func (c *Configurator) IncomingHTTPSConfig() *tls.Config {
 	return config
 }
 
-// OutgoingTLSConfigForCheck generates a *tls.Config for outgoing TLS connections
-// for checks. This function is separated because there is an extra flag to
-// consider for checks. EnableAgentTLSForChecks and InsecureSkipVerify has to
-// be checked for checks.
+// OutgoingTLSConfigForCheck creates a client *tls.Config for executing checks.
+// It is RECOMMENDED that the serverName be left unspecified. The crypto/tls
+// client will deduce the ServerName (for SNI) from the check address unless
+// it's an IP (RFC 6066, Section 3). However, there are two instances where
+// supplying a serverName is useful:
+//
+//  1. When the check address is an IP, a serverName can be supplied for SNI.
+//     Note: setting serverName will also override the hostname used to verify
+//     the certificate presented by the server being checked.
+//
+//  2. When the a hostname in the check address won't be present in the SAN
```

Comment:
`the a`? You mean `When the hostname in the check` right?

Reply:
Good catch!